main.js

```diff
@@ -101,22 +110,39 @@ function getRandomHex() {
     return (Math.floor(Math.random() * 16777216).toString(16).padStart(6, 0)).toUpperCase();
 }
 
-function savePalette() {
+async function savePalette() {
+    modalClassToggler([[saveModal, 'hidden'], [main, 'block'], [savedContainer, 'block']]);
     if (!savedPalettes.length) {
         paragraph.classList.add('hidden');
     } 
     if (isPaletteUnique(savedPalettes, currentPalette)) {
+        await getPromiseFromEvent2(saveModalButton, 'click');
         savedPalettes.push(currentPalette);
         addPaletteToSavedPalettes(currentPalette);
     }
+    modalClassToggler([[saveModal, 'hidden'], [main, 'block'], [savedContainer, 'block']]);
     getNewHexes(mainColorBoxes);
 }
 
-function isPaletteUnique(palettesList, singlePalette) {
-    for (i = 0; i < palettesList.length; i++) {
+function getPromiseFromEvent2(element, listenerName) {
+    return new Promise(function (resolve) {
+        function listener(event) {
+            if (event.target.nodeName === 'BUTTON') {
+                currentPalette.name = saveModalInput.value;
+                saveModalInput.value = ''
+            }
+            element.removeEventListener(listenerName, listener);
```

Comment:
I don't understand what the purpose of this is. It does not affect anything when I remove this line of code and it still works as intended.

Reply:
It sure affected it for me in my save modal functions!  If I removed this line, the promise chain broke - likely due to the recursion call/validation.